Refs #23677 - Add additional config options to foreman-debug

quba42 opened the pull request:
Allows for full control of upload behaviour of foreman-debug via the
config file. The following new variables are introduced: UPLOAD_URL,
UPLOAD_CMD, UPLOAD_USAGE, UPLOAD_MSG_UNSET, UPLOAD_MSG_SUCCESS,
and UPLOAD_MSG_FAIL.

https://projects.theforeman.org/issues/23677

<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` or `Refs #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Extract all strings for i18n, see [Translating section in the guide]
(https://projects.theforeman.org/projects/foreman/wiki/Translating)
* Prepend `[WIP]` for work in progress from bots triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* Be nice and respectful

We are running bots that will poke you if you miss an item from the list :-)

--->

quba42 committed:
Fixes #23677 - Add additional config options to foreman-debug

Allows for full control of upload behaviour of foreman-debug via the
config file. The following new variables are introduced: UPLOAD_URL,
UPLOAD_CMD, UPLOAD_USAGE_MSG, UPLOAD_UNSET_MSG, UPLOAD_SUCCESS_MSG,
and UPLOAD_FAIL_MSG.

lzap commented on config/foreman-debug.conf:
```diff
@@ -33,5 +33,38 @@
+#The tarball could not be uploaded, please upload it to an alternate location
+#and contact us on our mailing list or IRC referencing that URL.
+#UPLOAD_MSG_FAIL
+#)
```
Can you name all "message" variables in a same pattern? I'd prefer `MSG_*`. Also, I don't think we need to have copies of those in comments, just put "Custom message goes here" or similar.

quba42 replied:
I can do that. Will have a look at it tomorrow.

lzap reviewed (changes requested):
Hey, thanks. One nitpick, `UPLOAD_USAGE` should be tagged with `MSG`. I'd prefer prefix for all variables.

quba42 commented:
So, I added an additional fixup commit (for now), that (hopefully) addresses the requested changes.
_Edit: I squashed the fixup because it was getting messy, and to satisfy the foreman-bot._

- I removed the default values from the comments in the config file for any message variables in favor of "Add your custom message here."
- I also made all message variables as consistent as possible with respect to the need for trailing newline characters.
- I found a way to remove my useless use of cat while still having readable multi line strings in the script.
- All message variables are now named with a `_MSG` postfix part.

My arguments in favor of a `_MSG` postfix (rather than a `MSG_` prefix):

1. It keeps all the upload related variables consistent since they all start with `UPLOAD`.
2. It is more in line with a natural language reading of the message variable names. ("print the upload success message here").

If these arguments are not persuasive, I am happy to change things to a `MSG_` prefix one more time. (My preference for the posfix isn't that strong).

lzap commented on script/foreman-debug:
```diff
@@ -149,6 +144,22 @@ VERBOSE=0
+UPLOAD_UNSET_MSG="\
+To upload a tarball to our secure site, please use the -u option."
+UPLOAD_SUCCESS_MSG="\
+The tarball has been uploaded, please contact us on our mailing list or IRC
```
Mind changing this to `https://community.theforeman.org`? Even "IRC or Matrix" would be good to mention.

quba42 replied:
The Messages now read:
```
UPLOAD_SUCCESS_MSG="\
The tarball has been uploaded, please contact us on community.theforeman.org
or on IRC/Matrix (#theforeman), making sure you reference the following URL:

    http://debugs.theforeman.org/"
UPLOAD_FAIL_MSG="\
The tarball could not be uploaded, please upload it to an alternate location
and contact us on community.theforeman.org or on IRC/Matrix (#theforeman),
making sure you reference that URL."
```

ekohl replied:
https://theforeman.org/support.html#IRClivechat or https://theforeman.org/support.html#Matrix.orglivechat could be linked to for IRC/Matrix. Or just https://theforeman.org/support.html in general because it also refers to the forum.

lzap commented:
All is fine, one small commend plus can you rebase to `Fixes #23677 - ...` we use Refs only when we do additional fix into the same release. If you use Fixes our bot will automatically close the issue. Keep just one commit and it's ready to be tested and merged!

lzap reviewed (approved):
Thanks, I hope it will do the trick for you :)

quba42 commented:
Thanks!
